docs/sql-ref-datatypes.md

```diff
@@ -706,3 +708,53 @@ The following table shows the type names as well as aliases used in Spark SQL pa
 </table>
 </div>
 </div>
+
+### Floating Point Special Values
+
+Spark SQL supports several special floating point values in a case-insensitive manner:
+
+ * Inf/+Inf/Infinity/+Infinity: positive infinity
+   * ```FloatType```: 1.0f / 0.0f, which is equal to the value returned by <code>java.lang.Float.intBitsToFloat(0x7f800000)</code>.
+   * ```DoubleType```: 1.0 / 0.0, which is equal to the value returned by <code>java.lang.Double.longBitsToDouble(0x7ff0000000000000L)</code>.
+ * -Inf/-Infinity: negative infinity
+   * ```FloatType```: -1.0f / 0.0f, which is equal to the value returned by <code>java.lang.Float.intBitsToFloat(0xff800000)</code>.
+   * ```DoubleType```: -1.0 / 0.0, which is equal to the value returned by <code>java.lang.Double.longBitsToDouble(0xfff0000000000000L)</code>.
+ * NaN: not a number
+   * ```FloatType```: 0.0f / 0.0f, which is equivalent to the value returned by <code>java.lang.Float.intBitsToFloat(0x7fc00000)</code>.
+   * ```DoubleType```:  0.0d / 0.0, which is equivalent to the value returned by <code>java.lang.Double.longBitsToDouble(0x7ff8000000000000L)</code>.
+
+#### Examples
+
+{% highlight sql %}
+SELECT double('infinity');
+  +------------------------+
```

Comment:
nit: remove the leading space baesd on the @HyukjinKwon suggestion?

Reply:
Yeah, let's remove two leading spaces just to be consistent with other DBMS's doc references.

Reply:
I will remove the two leading spaces for this PR for now. 
I almost finished everything on Xiao's SQL reference list. After I am done, I will submit a PR to remove the two leading spaces in all the examples in SQL reference docs.

Reply:
Looks nice, thanks for the hard work, @huaxingao !